frame/contracts/src/wasm/prepare.rs

```diff
@@ -448,9 +455,14 @@ pub fn prepare_contract<C: ImportSatisfyCheck, T: Config>(
 		maximum: memory_limits.1,
 		_reserved: None,
 		code: contract_module.into_wasm_code()?,
+		original_code_len: original_code.len() as u32,
+		refcount: 1,
```

Comment:
This is called by code_cache when reinstrumenting with a new schedule and could reset the refcount, I think. See previous comment

Reply:
Good catch! Fixed it.